x-pack/docs/en/rest-api/security/query-api-key.asciidoc

```diff
@@ -0,0 +1,258 @@
+[role="xpack"]
+[[security-api-query-api-key]]
+=== Query API key information API
+++++
+<titleabbrev>Query API key information</titleabbrev>
+++++
+
+Retrieves information for API keys with <<query-dsl,Query DSL>>
+in a <<paginate-search-results,pagniated>> fashion.
+
+[[security-api-query-api-key-request]]
+==== {api-request-title}
+
+`GET /_security/_query/api_key`
+
+`POST /_security/_query/api_key`
+
+[[security-api-query-api-key-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_own_api_key` cluster privilege.
+
+[[security-api-query-api-key-desc]]
+==== {api-description-title}
+
+The information for the API keys created by
+<<security-api-create-api-key,create API Key>> can be retrieved using this API in
+a paginated manner and optionally filtered with a query.
+
+[[security-api-query-api-key-request-body]]
+==== {api-path-body-title}
+
+The following parameters can be specified in the body of a request and
+pertain to retrieving api keys:
+
+`query`::
+(Optional, string) A <<query-dsl,query>> to filter API keys to be returned.
+Note it only support a subset of the query types including
+<<query-dsl-match-all-query>>, <<query-dsl-bool-query>>,
+<<query-dsl-term-query>>, <<query-dsl-terms-query>>, <<query-dsl-ids-query>>,
+<<query-dsl-prefix-query>>, <<query-dsl-wildcard-query>>, <<query-dsl-range-query>>.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=from]
++
+By default, you cannot page through more than 10,000 hits using the `from` and
+`size` parameters. To page through more hits, use the
+<<search-after,`search_after`>> parameter.
+
+`size`::
+(Optional, integer) The number of hits to return. Needs to be non-negative.
++
+By default, you cannot page through more than 10,000 hits using the `from` and
+`size` parameters. To page through more hits, use the
+<<search-after,`search_after`>> parameter.
+
+`sort`::
+(Optional, object) <<sort-search-results,Sort definition>>. Other than `id`,
+all fields of an API key are eligible for sorting. In addition, sort can
+also be applied to the `_doc` field, which is useful for tie-breaking.
+
+`search_after`::
+(Optional, array) <<search-after,Search after>> definition.
+
+NOTE: If the user only has `manage_own_api_key` privilege, this API returns
+only those API keys that are owned by the user.
```

Comment:
I think we need to state the reverse as well.

> If the user has `manage_api_key` or higher (including `manage_security`), 
> this API returns API keys that are owned by any user

(any user? all users?)

Reply:
Good point. I added
> On the other hand, if the user
has `manage_api_key` or greater privilege (including `manage_security`), this
API returns API keys regardless of their owners.
?